File: swift/0229. Majority Element II.swift
class Solution {
    func majorityElement(_ nums: [Int]) -> [Int] {
        let threshold = nums.count / 3

        var numCount = [Int:Int]()
        var set = Set<Int>() 

        for num in nums {
            numCount[num, default:0] += 1
            if numCount[num]! > threshold {
                set.insert(num)
            }
        }

        return Array(set)
    }
}

File: python/0229. Majority Element II.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
            
        map = defaultdict(int)
        k = len(nums) // 3
        ans = []

        for num in nums:
            map[num] += 1
            if map[num] == k+1:
                ans.append(num)

        return ans
